### PR TITLE
[Backend] PATCH/PUT API for updating the isApproved field of Question & Answers stored in Db

### PR DESCRIPTION
**Pull request by arinmodi**
## Issue that this pull request solves
 Closes: #730

## Proposed changes

- Create an API for updating the isApproved field of questions data [the api will resider under [here](https://github.com/HITK-TECH-Community/Community-Website/tree/main/backend/app/routes/Q%26A/question)]
- Create an API for updating the isApproved field of answers data [The api will reside under [here](https://github.com/HITK-TECH-Community/Community-Website/tree/main/backend/app/routes/Q%26A/answers)]
- Write tests for both of them

### Brief description of what is fixed or changed

Added Two Patch Files,
1) updateQuestionStatus
2) updateAnswerStatus

- Did corresponding changes in the routes file also added the corresponding tests

### ScreenShots

- Update Answer Status:

image

- Update Question Status:

image

## Types of changes

_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Documentation content changed)
- [ ] Other (please describe): 

## Checklist

_Put an `x` in the boxes that apply_

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] My changes does not break the current system and it passes all the current test cases.

### CODE REVIEW

**Comment by arinmodi** [comment]
@Kajol-Kumari, please review this and give update.

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
Hi @arinmodi 
Thank you for taking time to work on this. Can you please do below things -

1. Please remove all the binary file. You don't need to push the `upload` folder
2. Please test the apo via postman and attach the screenshot in the pr description

**Comment by arinmodi** [comment]
@Kajol-Kumari, Changes done

File: backend/app/routes/Q&A/question/updateQuestionStatus.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            message: 'Database Error',
+            errStack: err,
+        });
+
+        return next(Error);
+    }
+
+    // if result is null that means question with given id is not exist in collection
+    if(result === null) {
+        const Error2 = new ErrorHandler(constants.ERRORS.INPUT, {
```
Comment (Kajol-Kumari) [inline backend/app/routes/Q&A/question/updateQuestionStatus.js:29]: Can we please give meaningful name here? Also, please use `cameCase` for variable name like `error`

**Review by Kajol-Kumari — APPROVED** [review]
LGTM 👍 
It would be great if you can update it with the neat changes requested :)

File: backend/app/routes/Q&A/question/updateQuestionStatus.js (hunk 1 above)
Comment (arinmodi) [reply backend/app/routes/Q&A/question/updateQuestionStatus.js]: Fixed

**Comment by Kajol-Kumari** [comment]
> @Kajol-Kumari, who will do second review ?

Please let me know if any specific label is needed for being counted under hacktoberfest

**Comment by arinmodi** [comment]
<b>hacktoberfest</b> label is needed on this issue, 

Thanks for giving reminder, I completely forgot about that

**Comment by arinmodi** [comment]
@Kajol-Kumari, can you please add the <b>hacktoberfest</b> label on this issue ?

**Pull request merged**